src/matrix.jl

```diff
@@ -70,39 +83,59 @@ function LinearAlgebra.opnorm(A::IntervalMatrix, p::Real=Inf)
 end
 
 """
-    left(A::IntervalMatrix)
+    left(A::IntervalMatrix{T}) where {T}
 
-The left part of this interval matrix, which corresponds to the `left` of
-each interval element in the matrix.
+Return the left part of this interval matrix, which corresponds to taking the
+element-wise infimum of `A`.
 
 ### Input
 
 - `A` -- interval matrix
+
+### Output
+
+A scalar matrix whose coefficients are the infima of each element in `A`.
 """
-function left(A::IntervalMatrix)
-    n = size(A, 1)
-    return hcat([[A[i, j].left for j in 1:n] for i in 1:n]...)'
+function left(A::IntervalMatrix{T}) where {T}
+    m, n = size(A)
+    L = Matrix{T}(undef, m, n)
+    for j in 1:m
+        for i in 1:n
+            @inbounds L[i, j] = inf(A[i, j])
+        end
+    end
+    return L
 end
 
 """
-    right(A::IntervalMatrix)
+    right(A::IntervalMatrix{T}) where {T}
 
-The right part of this interval matrix, which corresponds to the `right` of
-each interval element in the matrix.
+Return the right part of this interval matrix, which corresponds to taking the
+element-wise supremum of `A`.
 
 ### Input
 
 - `A` -- interval matrix
+
+### Output
+
+A scalar matrix whose coefficients are the suprema of each element in `A`.
 """
-function right(A::IntervalMatrix)
-    n = size(A, 1)
-    return hcat([[A[i, j].right for j in 1:n] for i in 1:n]...)'
+function right(A::IntervalMatrix{T}) where {T}
```

Comment:
I think it's good as is, such simple stuff.

Reply:
There is a bug in your *simple stuff* (you confused `m` and `n`). Maybe you want to change it now? :stuck_out_tongue:

Reply:
Thanks for pointing it out, i can make the change in each function. The bug is in `master`, since `left` and `right` are assuming that the given matrix is square, but it need not be the case (https://github.com/JuliaReach/IntervalMatrices.jl/blob/master/src/matrix.jl#L82).

Reply:
I could not resist and added a `map` method. This could also be useful for other means in the future.

Reply:
Since `IntervalMatrix` is an abstract array, Julia's built-in `map` function "just works".

Reply:
Indeed, but it seems a bit slower (probably because [the fallback is implemented in a different way](https://github.com/JuliaLang/julia/blob/5dd7636b9033a48853edb5c6b41c06bc3975ecf1/base/abstractarray.jl#L2073), but it could also just be noise in the measurements).

```julia
julia> using IntervalMatrices, BenchmarkTools
julia> m = IntervalMatrix([-1.1..0.9 -4.1.. -3.9; 3.9..4.1 -1.1..0.9]);
julia> @btime map(x->x^2, m);
  8.918 μs (162 allocations: 6.67 KiB)  # fallback
  8.607 μs (162 allocations: 6.67 KiB)  # my method
```
Still, you can drop the commit.